### PR TITLE
Security-CVEs fixes guidelines

### PR DESCRIPTION
**Pull request by himsgupta1122**
### Description
Any CVE package upgrade may be miss-handled by over-looking sub dependency update which keeps the vulnerability. The above addition provides a HOW-TO and hence reduces the burden on the maintainers.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
Hello @himsgupta1122 thanks for adding this.

Could you update your commits to include a signed off by. Like `git commit -s`.

Thank you!

File: SECURITY.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
\ No newline at end of file
+- If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/) or directly via email to aws-security@amazon.com. Please do **not** create a public GitHub issue.
+
+- For Security-CVE related fix - 
+    - For direct dependency - Use ```yarn upgrade package``` to update the package and in order to enforce as sub-deps please add nested-dep step2.
+
+    - For nested dependency/sub-deps - In order to enforce package above Vx.y.z, we can add version in the resolutions [section](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) for all the package sub-deps or specific package sub-dep. For more on version updates please see 
+[Why](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-why-would-you-want-to-do-this) and [How](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-how-to-use-it) to upgrade.
```
Comment (seraphjiang) [inline SECURITY.md:9]: could we also add example code for this case

**Comment by seraphjiang** [comment]
@kavilla where is right place to add such guideline which audience is dashboards developer. 
should it go to developer guideline or security guidline

File: SECURITY.md (hunk 1 above)
Comment (himsgupta1122) [reply SECURITY.md:9]: sure.

**Comment by himsgupta1122** [comment]
> @kavilla where is right place to add such guideline which audience is dashboards developer. should it go to developer guideline or security guidline

@kavilla, any suggestions here.

**Review by zhongnansu — COMMENTED** [review]
Can you provide more context? Better create an issue for it, explaining why do we need a guideline, why the current process is not clear to people who wants to fix CVEs

**Comment by himsgupta1122** [comment]
> Can you provide more context? Better create an issue for it, explaining why do we need a guideline, why the current process is not clear to people who wants to fix CVEs

Context: Any CVE package upgrade may be miss-handled by over-looking sub dependency update which keeps the vulnerability. The above addition provides a HOW-TO and hence reduces the burden on the maintainers. @zhongnansu

**Comment by zhongnansu** [comment]
> > Can you provide more context? Better create an issue for it, explaining why do we need a guideline, why the current process is not clear to people who wants to fix CVEs
> 
> Context: Any CVE package upgrade may be miss-handled by over-looking sub dependency update which keeps the vulnerability. The above addition provides a HOW-TO and hence reduces the burden on the maintainers. @zhongnansu

got it, can you add to the PR description?

File: SECURITY.md [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+      For direct dependency checks: 
+              run: yarn upgrade foobar@1.y 
+      Step 2.
+      Check for sub deps foobar in other package. 
+              If foobar@1.x exists for subdeps in yarn.lock file
+              Then edit the package.json file and add **/foobar@1.y in resolution section as shown below to enforce the 1.y.
+                'resolutions': { "**/foobar": "^1.y", 
+                                 "**/foo": "^2.x" ,
+                                 "**/bar": "^3.k"}
+
```
Comment (zhongnansu) [inline SECURITY.md:22]: Can add instruction on how to backport to needed branch, `backport labels`. A cve fix in main will only take effect when it's backported to corresponding versions, right? The flow is not complete if backport is not done.

File: SECURITY.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
 ## Reporting a Vulnerability
 
-If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/) or directly via email to aws-security@amazon.com. Please do **not** create a public GitHub issue.
\ No newline at end of file
+- If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/) or directly via email to aws-security@amazon.com. Please do **not** create a public GitHub issue.
+
+- For Security-CVE related fix - 
+    - For direct dependency - Use ```yarn upgrade package``` to update the package and in order to enforce as sub-deps please add nested-dep step2.
```
Comment (zhongnansu) [inline SECURITY.md:6]: better to be clear that this command updates `yarn.lock` file?

File: SECURITY.md [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+
+    - For nested dependency/sub-deps - In order to enforce package above Vx.y.z, we can add version in the resolutions [section](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) for all the package sub-deps or specific package sub-dep. For more on version updates please see 
+[Why](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-why-would-you-want-to-do-this) and [How](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/#toc-how-to-use-it) to upgrade.
+       ```
+      Example: foobar@1.x vulnerable package and 1.y is the fix 
+      step 1: 
+      For direct dependency checks: 
+              run: yarn upgrade foobar@1.y 
+      Step 2.
+      Check for sub deps foobar in other package.
```
Comment (zhongnansu) [inline SECURITY.md:16]: I don't think just by changing this resolution section is enough. You still needs to run some other yarn commands that will change the `yarn.lock` file. I am not sure which exact command is it, is it `yarn install`?

**Review by kristenTian — APPROVED** [review]

**Pull request merged**